tests/test_shell.py

```diff
@@ -1,30 +1,180 @@
 import os
-import unittest
+import sys
+from datetime import datetime, timedelta
 
+import pytest
+
+from unfurl.configurator import Status
+from unfurl.configurators.shell import ShellConfigurator, subprocess
 from unfurl.job import JobOptions, Runner
 from unfurl.yamlmanifest import YamlManifest
 
 
-class ShellConfiguratorTest(unittest.TestCase):
-    def setUp(self):
+class TestShellConfigurator:
+    def setup(self):
         path = os.path.join(
             os.path.dirname(__file__), "examples", "shell-ensemble.yaml"
         )
         with open(path) as f:
-            self.manifest = f.read()
+            self.ensemble = f.read()
 
     def test_shell(self):
         """
         test that runner figures out the proper tasks to run
         """
-        runner = Runner(YamlManifest(self.manifest))
-
-        run1 = runner.run(JobOptions(instance="test1"))
-        assert len(run1.workDone) == 1, run1.workDone
-        self.assertEqual(
-            runner.manifest.getRootResource()
-            .findResource("test1")
-            .attributes["stdout"],
-            "helloworld",
+        runner = Runner(YamlManifest(self.ensemble))
+
+        job = runner.run(JobOptions(instance="test1"))
+
+        assert len(job.workDone) == 1, job.workDone
+        assert (
+            runner.manifest.getRootResource().findResource("test1").attributes["stdout"]
+            == "helloworld"
         )
-        assert not run1.unexpectedAbort, run1.unexpectedAbort.getStackTrace()
+        assert not job.unexpectedAbort, job.unexpectedAbort.getStackTrace()
+
+    def test_timeout(self):
```

Comment:
@jozo does this need to be skipped < 3.7 like the following test?

Reply:
Not needed. Older versions of Python also raise TimeoutExpired exception. That is correct. But they do it after 15 seconds instead of just 1 second. I'll change it to `sleep 5` to save few seconds from duration of tests.